apps/dsb-message-broker/test/channel.e2e-spec.ts

```diff
@@ -52,28 +82,177 @@ describe('ChannelController (e2e)', () => {
+            .patch('/channel')
+            .send({
+                fqcn,
+                publishers: [authenticatedUser3.did],
+                subscribers: [authenticatedUser3.did]
+            })
+            .set('User-No', '3')
+            .expect(HttpStatus.UNAUTHORIZED);
+    });
 
-    // it('should not create a channel with missing fqcn', async () => {});
+    it('should modify a channel', async () => {
+        await request(app)
+            .patch('/channel')
+            .send({
+                fqcn,
+                publishers: [authenticatedUser3.did],
+                subscribers: [authenticatedUser3.did]
+            })
+            .set('User-No', '2')
+            .expect(HttpStatus.OK);
+    });
 
-    // it('should not create a channel with invalid fqcn', async () => {});
+    it('should not be able to get accessible channels without having user role', async () => {
+        await request(app).get('/channel/pubsub').set('User-No', '4').expect(HttpStatus.FORBIDDEN);
+    });
 
-    // it('should not create a channel with invalid metadata', async () => {});
+    it('should not be able to get channel metadata without being publisher or subscriber', async () => {
+        await request(app)
+            .get(`/channel/${fqcn}`)
+            .set('User-No', '2')
+            .expect(HttpStatus.UNAUTHORIZED);
+    });
 
-    // it('should modify a channel', async () => {});
+    it('should get accessible channels', async () => {
+        const channels = (
+            await request(app).get('/channel/pubsub').set('User-No', '3').expect(HttpStatus.OK)
+        ).body;
 
-    // it('should not be able to modify a channel without having admin permission', async () => {});
+        const includesCreatedChannel = channels.some((channel: any) => channel.fqcn === fqcn);
+
+        expect(includesCreatedChannel, 'it should include created channel').to.be.true;
+    });
 
-    // it('should remove a channel', async () => {});
+    it('should get channel metadata', async () => {
+        const channel = (
+            await request(app).get(`/channel/${fqcn}`).set('User-No', '3').expect(HttpStatus.OK)
+        ).body;
 
-    // it('should not be able to remove a channel without having admin permission', async () => {});
+        expect(channel).to.include({ fqcn }, 'fqcn should match');
+
+        expect(channel).to.deep.include(
+            { admins: ['did:ethr:0x46646c919278e1Dac6ef3B02BC520A82B8FaA596'] },
+            'should have creator as admin'
+        );
+
+        expect(channel).to.deep.include(
+            { publishers: ['did:ethr:0x46646c919278e1Dac6ef3B02BC520A82B8FaA237'] },
+            'should have modified publishers'
+        );
+
+        expect(channel).to.deep.include(
+            { subscribers: ['did:ethr:0x46646c919278e1Dac6ef3B02BC520A82B8FaA237'] },
+            'should have modified subscribers'
+        );
+    });
+
+    it('should not be able to remove a channel without having user role', async () => {
```

Comment:
`should not be able to remove a channel when having user role`

Reply:
Or actually what's the use case about?

Reply:
addtional to be admin of the channel to remove, user should have user role to do that

https://github.com/energywebfoundation/dsb/blob/78eb1f0fefda62f755b8d1a7ef15f1413fe708b2/apps/dsb-message-broker/src/channel/channel.controller.ts#L159